src/lib/api/types.d.ts

```diff
@@ -33,11 +33,18 @@ export type ViewerMode = ReadMode | WriteMode;
 
 export type Zoom = number | Sizes | "width" | "height";
 
-export interface APIError {
-  error: {
-    status: number;
-    message: string;
-  };
+export interface APIError<E> {
+  status: number;
+  message: string;
+  errors?: E;
+}
+
+/**
+ * Wrap an API response so we can pass errors along
+ */
+export interface APIResponse<T, E> {
```

Comment:
```suggestion
export interface APIResponse<D, E> {
```

Purely style, but popular convention is for the generic type to be the initial of the property it's associated with.